src/parse_tree.rs

```diff
@@ -127,7 +145,8 @@ impl Root {
 		}
 		
 		if !expr_stack.is_empty() {
-			todo!("geeigneten Fehlercode zurückgeben")
+			//Gar nicht sicher welcher Typ von Error hier sein muss
+			return Err(Error::Semantic);
```

Comment:
If the stack is not empty after all tokens have been matched this means that there is more than one expression on the stack and therefore a syntactical error 

> since there will be another expression that cannot be used e.g. 123+-***** or 12+**1** => syntax error).

This is because the above implementation recursively adds the latest operation on top of the stack and wraps another operation as the right hand statement as an expression.

**Stack:** 
```rust
// input: 123*+
 [Expr(Expr::Add(Expr::Int(1), Expr(Expr::Mul(Expr::Int(2), Expr::Int(3)))))]
// output: 1 + 2 * 3
```

 Therefore there will only be one expression left in the stack, no matter how many operations you use.

Reply:
very good explained